### ISSUE TITLE
error in reverse_words beam search?

### ISSUE THREAD

**Issue by chrishokamp** [mila-udem/blocks-examples#21]
Hi all, 

First of all, thanks for the incredible work! I was running the reverse_words example, and everything works, but when I run:

`python -m reverse_words beam_search  <path_to_my_model>`

I get an error:
```
Traceback (most recent call last):
  File "/ichec/home/users/chokamp/programs/anaconda/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/ichec/home/users/chokamp/programs/anaconda/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/ichec/home/users/chokamp/projects/blocks-examples/reverse_words/__main__.py", line 35, in <module>
    main(**vars(args))
  File "reverse_words/__init__.py", line 327, in main
    batch_size, axis=1))
  File "reverse_words/__init__.py", line 285, in generate
    ComputationGraph(generated[1]))
ValueError: too many values to unpack
```

After digging around a bit, I can fix it by changing [these lines](https://github.com/mila-udem/blocks-examples/blob/master/reverse_words/__init__.py#L273-L275) to:

```
                (initial_states, samples) = VariableFilter(
                    bricks=[reverser.generator], name="outputs")(
                        ComputationGraph(generated[1]))
```

I.e. adding the `intial_states` so that `samples` is the correct variable. I'd love to submit a pull request but wanted to check first if there's anything I'm missing.

Cheers!

**Comment by rizar** [comment]
Thanks for reporting this issue! It is a consequence of a recent Blocks refactoring. A better fix would be to follow the fix [in beam search test](https://github.com/mila-udem/blocks/blob/master/tests/test_search.py#L92), that is instead of filtering by the brick filtering by the application should be used. 

A PR would be appreciated, and if you want to do it really, really nice, we should test the beam search, at least with random parameters. Currently we only test training.

**Comment by rizar** [comment]
Oops, forgot about that one. Just fixed this issue in #47
